pkg/services/librarypanels/api.go

```diff
@@ -0,0 +1,31 @@
+package librarypanels
+
+import (
+	"github.com/go-macaron/binding"
+	"github.com/grafana/grafana/pkg/api"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func (lps *LibraryPanelService) registerAPIEndpoints() {
+	if !lps.IsEnabled() {
+		return
+	}
+
+	lps.RouteRegister.Group("/api/library-panels", func(libraryPanels routing.RouteRegister) {
+		libraryPanels.Post("/", middleware.ReqSignedIn, binding.Bind(addLibraryPanelCommand{}), api.Wrap(lps.createHandler))
+	})
+}
+
+// createHandler handles POST /api/library-panels.
+func (lps *LibraryPanelService) createHandler(c *models.ReqContext, cmd addLibraryPanelCommand) api.Response {
+	panel, err := lps.createLibraryPanel(c, cmd)
+
+	if err != nil {
+		return api.Error(400, "Failed to create library panel", err)
+	}
+
+	return api.JSON(200, util.DynMap{"panel": panel})
+}
```

Comment:
I've noticed that we usually add api handlers as methods on our entities and not as functions. Is there any specific reasons for this? I'm guessing convenience @papagian @aknuds1 but I'm curious to why?

Reply:
So instead of 

```go
func (lps *LibraryPanelService) registerAPIEndpoints()
```

Why not

```go
func registerAPIEndpoints(lps *LibraryPanelService)
```

Reply:
How would the handler have access to `lps.createLibraryPanel` without the method receiver, i.e. `lps`?

Reply:
You would have to pass lps *LibraryPanelService to the handler I guess?

Reply:
Ah, I see you're asking if it should be a free function receiving `lps`. I guess you could say that about any method, it could receive the object as a normal argument instead? I would say it's just about what makes sense logically. If it's working on `lps`, why not make it a method?

Reply:
I'm just trying to understand how we define things around here :) I mean `createLibraryPanel` could also be a "free" function and then there is no reason to pass lps at all. I'm just trying to figure out when to define methods (on/for structs) and when to define "free" functions. 

I got the question from https://www.youtube.com/watch?v=29LLRKIL_TI&feature=youtu.be&t=891 when I was digging deeper into pointers & values. 

So to sum it up:
- use what makes most sense logically
- prefer methods over functions to minimize pollution of namespaces

Thank you again for all the help @aknuds1 @papagian @torkelo